api/dataprocessor.go

```diff
@@ -621,7 +621,14 @@ func mergeSeries(in []models.Series) []models.Series {
 		if len(series) == 1 {
 			merged[i] = series[0]
 		} else {
-			//we use the first series in the list as our result.  We check over every
+			// TODO
+			// Remove this when https://github.com/grafana/metrictank/issues/913 is fixed
+			for i := range series {
```

Comment:
could this be a range over keys/values and avoid having to reindex series[i]?

Reply:
Values are copies in go with refs to underlying pointers. So, if `Fix` reallocates `Datapoints` then the original wouldn't get the update. technically, I could get a name for just passing the params to `Fix`